Unit Tests/Tests/Buffer/ConcreteBufferTests.swift

```diff
@@ -0,0 +1,619 @@
+        /******************/
+        
+        let initialState: VersionedState = .initial
+        
+        let stubStore = StubStore(state_toReturn: initialState, register_expectedCallCount: 1, unregister_expectedCallCount: 1)
+        let stubFilter = StubStateFilter(hasCompleteSubstate_defaultValueToReturn: true)
+        
+        let dependencies = DependenciesDoubles(store: stubStore)
+        
+        /******************/
+        /*----- WHEN -----*/
+        /******************/
+        
+        let sut = ConcreteBuffer(filter: stubFilter, dependencies: dependencies)
+        
+        /******************/
+        /*----- THEN -----*/
+        /******************/
+        
+        let expectedState = initialState
+        
+        XCTAssertEqual(sut.currentState, expectedState)
+    }
+    
+    func test_init_registersAsStoreListener() {
+        
+        /******************/
+        /*---- GIVEN -----*/
+        /******************/
+        
+        let initialState: VersionedState = .initial
+        
+        let stubStore = StubStore(state_toReturn: initialState, register_expectedCallCount: 1, unregister_expectedCallCount: 1)
+        let stubFilter = StubStateFilter(hasCompleteSubstate_defaultValueToReturn: true)
+        
+        let dependencies = DependenciesDoubles(store: stubStore)
+        
+        /******************/
+        /*----- WHEN -----*/
+        /******************/
+        
+        let sut = ConcreteBuffer(filter: stubFilter, dependencies: dependencies)
+        
+        /******************/
+        /*----- THEN -----*/
+        /******************/
+        
+        XCTAssertEqual(stubStore.register_actualCallCount, 1)
+        XCTAssertTrue(stubStore.register_listenerLastReceived === sut)
+    }
+    
+    func test_init_withInitialStateThatsCompleteWithRelevantSignature_currentStateIsSet() {
+        
+        /******************/
+        /*---- GIVEN -----*/
+        /******************/
+        
+        let initialState = VersionedState(
+            chatState: .empty,
+            auxiliaryState: .empty,
+            version: 1,
+            signature: .initialState
+        )
+        
+        let stubStore = StubStore(state_toReturn: initialState, register_expectedCallCount: 1, unregister_expectedCallCount: 1)
+        let stubFilter = StubStateFilter(hasModifiedSubstate_defaultValueToReturn: true,
+                                         hasCompleteSubstate_defaultValueToReturn: true,
+                                         hasRelevantSignature_defaultValueToReturn: true)
+        let stubDelegate = StubBufferDelegate(didUpdateState_expectedCallCount: 0)
+        
+        let dependencies = DependenciesDoubles(store: stubStore)
+        
+        let sut = ConcreteBuffer(filter: stubFilter, dependencies: dependencies)
+        sut.delegate = stubDelegate
+        
+        /******************/
+        /*----- WHEN -----*/
+        /******************/
```

Comment:
Do these 4 new tests need updating as follows?...

```suggestion
        /******************/
        /*----- WHEN -----*/
        /******************/
        
        let sut = ConcreteBuffer(filter: stubFilter, dependencies: dependencies)
        sut.delegate = stubDelegate
```

Reply:
I've moved the code to `WHEN` section.